src/OrleansLoggerProvider/LegacyOrleansLogger.cs

```diff
@@ -0,0 +1,149 @@
+        {
+            this.logConsumers = logConsumers;
+            this.flushableConsumers = logConsumers.OfType<IFlushableLogConsumer>().ToList();
+            this.name = categoryName;
+            this.loggerType = DetermineLoggerType(categoryName);
+            this.ipEndPoint = ipEndPoint;
+        }
+
+        /// <inheritdoc/>
+        public IDisposable BeginScope<TState>(TState state)
+        {
+            if (state == null)
+            {
+                throw new ArgumentNullException(nameof(state));
+            }
+            //TODO: support logging scope
+            return NullScope.Instance;
+        }
+
+        /// <inheritdoc/>
+        public bool IsEnabled(LogLevel logLevel)
+        {
+            return true;
+        }
+
+        /// <summary>
+        /// Log a message. Current logger supports legacy event bulking feature. Message bulking feature will only log eventId code appearance count
+        /// if certain event appear more than <see cref="EventBulkingConfig.BulkEventLimit"/>> in <see cref="EventBulkingConfig.BulkEventInterval"/>
+        /// </summary>
+        /// <typeparam name="TState"></typeparam>
+        /// <param name="logLevel"></param>
+        /// <param name="eventId"></param>
+        /// <param name="state"></param>
+        /// <param name="exception"></param>
+        /// <param name="formatter"></param>
+        public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter)
+        {
+            var errorCode = eventId.Id;
+            var severity = LogLevelToSeverity(logLevel);
+            WriteLogMessageToLogConsumers(errorCode, severity, formatter(state, exception), exception);
+        }
+
+        /// <summary>
+        /// Map LogLevel to Severity
+        /// </summary>
+        /// <param name="logLevel"></param>
+        /// <returns></returns>
+        public static Severity LogLevelToSeverity(LogLevel logLevel)
+        {
+            switch (logLevel)
+            {
+                case LogLevel.None: return Severity.Off;
+                case LogLevel.Critical: return Severity.Error;
+                case LogLevel.Error: return Severity.Error;
+                case LogLevel.Warning: return Severity.Warning;
+                case LogLevel.Information: return Severity.Info;
+                case LogLevel.Debug: return Severity.Verbose;
+                default: return Severity.Verbose3;
+            }
+        }
+
+        private LoggerType DetermineLoggerType(string category)
+        {
+            LoggerType type = LoggerType.Application;
+            if (category.Contains("Orleans"))
+            {
+                type = LoggerType.Runtime;
+                if (category.Contains("Provider"))
+                    type = LoggerType.Provider;
+                if (category.Contains("Grain"))
```

Comment:
These can overwrite each other, no need for ```else``` ?

Reply:
it is intended this way. it basically means, if category contains Orleans and Provider than it is LoggerType.provider